File: sample-java/src/main/java/com/smart/github/domain/model/Trend.java
package com.smart.github.domain.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Trend {

    @SerializedName("id")
    @Expose
    private Integer id;
    @SerializedName("node_id")
    @Expose
    private String nodeId;
    @SerializedName("name")
    @Expose
    private String name;
    @SerializedName("full_name")
    @Expose
    private String fullName;
    @SerializedName("owner")
    @Expose
    private Owner owner;
    @SerializedName("private")
    @Expose
    private Boolean isPrivate;
    @SerializedName("html_url")
    @Expose
    private String htmlUrl;
    @SerializedName("description")
    @Expose
    private String description;
    @SerializedName("fork")
    @Expose
    private Boolean fork;
    @SerializedName("url")
    @Expose
    private String url;
    @SerializedName("archive_url")
    @Expose
    private String archiveUrl;
    @SerializedName("assignees_url")
    @Expose
    private String assigneesUrl;
    @SerializedName("blobs_url")
    @Expose
    private String blobsUrl;
    @SerializedName("branches_url")
    @Expose
    private String branchesUrl;
    @SerializedName("collaborators_url")
    @Expose
    private String collaboratorsUrl;
    @SerializedName("comments_url")
    @Expose
    private String commentsUrl;
    @SerializedName("commits_url")
    @Expose
    private String commitsUrl;
    @SerializedName("compare_url")
    @Expose
    private String compareUrl;
    @SerializedName("contents_url")
    @Expose
    private String contentsUrl;
    @SerializedName("contributors_url")
    @Expose
    private String contributorsUrl;
    @SerializedName("deployments_url")
    @Expose
    private String deploymentsUrl;
    @SerializedName("downloads_url")
    @Expose
    private String downloadsUrl;
    @SerializedName("events_url")
    @Expose
    private String eventsUrl;
    @SerializedName("forks_url")
    @Expose
    private String forksUrl;
    @SerializedName("git_commits_url")
    @Expose
    private String gitCommitsUrl;
    @SerializedName("git_refs_url")
    @Expose
    private String gitRefsUrl;
    @SerializedName("git_tags_url")
    @Expose
    private String gitTagsUrl;
    @SerializedName("git_url")
    @Expose
    private String gitUrl;
    @SerializedName("issue_comment_url")
    @Expose
    private String issueCommentUrl;
    @SerializedName("issue_events_url")
    @Expose
    private String issueEventsUrl;
    @SerializedName("issues_url")
    @Expose
    private String issuesUrl;
    @SerializedName("keys_url")
    @Expose
    private String keysUrl;
    @SerializedName("labels_url")
    @Expose
    private String labelsUrl;
    @SerializedName("languages_url")
    @Expose
    private String languagesUrl;
    @SerializedName("merges_url")
    @Expose
    private String mergesUrl;
    @SerializedName("milestones_url")
    @Expose
    private String milestonesUrl;
    @SerializedName("notifications_url")
    @Expose
    private String notificationsUrl;
    @SerializedName("pulls_url")
    @Expose
    private String pullsUrl;
    @SerializedName("releases_url")
    @Expose
    private String releasesUrl;
    @SerializedName("ssh_url")
    @Expose
    private String sshUrl;
    @SerializedName("stargazers_url")
    @Expose
    private String stargazersUrl;
    @SerializedName("statuses_url")
    @Expose
    private String statusesUrl;
    @SerializedName("subscribers_url")
    @Expose
    private String subscribersUrl;
    @SerializedName("subscription_url")
    @Expose
    private String subscriptionUrl;
    @SerializedName("tags_url")
    @Expose
    private String tagsUrl;
    @SerializedName("teams_url")
    @Expose
    private String teamsUrl;
    @SerializedName("trees_url")
    @Expose
    private String treesUrl;

    public Integer getId() {
        return id;
    }

    public String getNodeId() {
        return nodeId;
    }

    public String getName() {
        return name;
    }

    public String getFullName() {
        return fullName;
    }

    public Owner getOwner() {
        return owner;
    }

    public Boolean getPrivate() {
        return isPrivate;
    }

    public String getHtmlUrl() {
        return htmlUrl;
    }

    public String getDescription() {
        return description;
    }

    public Boolean getFork() {
        return fork;
    }

    public String getUrl() {
        return url;
    }

    public String getArchiveUrl() {
        return archiveUrl;
    }

    public String getAssigneesUrl() {
        return assigneesUrl;
    }

    public String getBlobsUrl() {
        return blobsUrl;
    }

    public String getBranchesUrl() {
        return branchesUrl;
    }

    public String getCollaboratorsUrl() {
        return collaboratorsUrl;
    }

    public String getCommentsUrl() {
        return commentsUrl;
    }

    public String getCommitsUrl() {
        return commitsUrl;
    }

    public String getCompareUrl() {
        return compareUrl;
    }

    public String getContentsUrl() {
        return contentsUrl;
    }

    public String getContributorsUrl() {
        return contributorsUrl;
    }

    public String getDeploymentsUrl() {
        return deploymentsUrl;
    }

    public String getDownloadsUrl() {
        return downloadsUrl;
    }

    public String getEventsUrl() {
        return eventsUrl;
    }

    public String getForksUrl() {
        return forksUrl;
    }

    public String getGitCommitsUrl() {
        return gitCommitsUrl;
    }

    public String getGitRefsUrl() {
        return gitRefsUrl;
    }

    public String getGitTagsUrl() {
        return gitTagsUrl;
    }

    public String getGitUrl() {
        return gitUrl;
    }

    public String getIssueCommentUrl() {
        return issueCommentUrl;
    }

    public String getIssueEventsUrl() {
        return issueEventsUrl;
    }

    public String getIssuesUrl() {
        return issuesUrl;
    }

    public String getKeysUrl() {
        return keysUrl;
    }

    public String getLabelsUrl() {
        return labelsUrl;
    }

    public String getLanguagesUrl() {
        return languagesUrl;
    }

    public String getMergesUrl() {
        return mergesUrl;
    }

    public String getMilestonesUrl() {
        return milestonesUrl;
    }

    public String getNotificationsUrl() {
        return notificationsUrl;
    }

    public String getPullsUrl() {
        return pullsUrl;
    }

    public String getReleasesUrl() {
        return releasesUrl;
    }

    public String getSshUrl() {
        return sshUrl;
    }

    public String getStargazersUrl() {
        return stargazersUrl;
    }

    public String getStatusesUrl() {
        return statusesUrl;
    }

    public String getSubscribersUrl() {
        return subscribersUrl;
    }

    public String getSubscriptionUrl() {
        return subscriptionUrl;
    }

    public String getTagsUrl() {
        return tagsUrl;
    }

    public String getTeamsUrl() {
        return teamsUrl;
    }

    public String getTreesUrl() {
        return treesUrl;
    }

    @Override
    public String toString() {
        return "Trend{" +
                "id=" + id +
                ", nodeId='" + nodeId + '\'' +
                ", name='" + name + '\'' +
                ", fullName='" + fullName + '\'' +
                ", owner=" + owner +
                ", isPrivate=" + isPrivate +
                ", htmlUrl='" + htmlUrl + '\'' +
                ", description='" + description + '\'' +
                ", fork=" + fork +
                ", url='" + url + '\'' +
                ", archiveUrl='" + archiveUrl + '\'' +
                ", assigneesUrl='" + assigneesUrl + '\'' +
                ", blobsUrl='" + blobsUrl + '\'' +
                ", branchesUrl='" + branchesUrl + '\'' +
                ", collaboratorsUrl='" + collaboratorsUrl + '\'' +
                ", commentsUrl='" + commentsUrl + '\'' +
                ", commitsUrl='" + commitsUrl + '\'' +
                ", compareUrl='" + compareUrl + '\'' +
                ", contentsUrl='" + contentsUrl + '\'' +
                ", contributorsUrl='" + contributorsUrl + '\'' +
                ", deploymentsUrl='" + deploymentsUrl + '\'' +
                ", downloadsUrl='" + downloadsUrl + '\'' +
                ", eventsUrl='" + eventsUrl + '\'' +
                ", forksUrl='" + forksUrl + '\'' +
                ", gitCommitsUrl='" + gitCommitsUrl + '\'' +
                ", gitRefsUrl='" + gitRefsUrl + '\'' +
                ", gitTagsUrl='" + gitTagsUrl + '\'' +
                ", gitUrl='" + gitUrl + '\'' +
                ", issueCommentUrl='" + issueCommentUrl + '\'' +
                ", issueEventsUrl='" + issueEventsUrl + '\'' +
                ", issuesUrl='" + issuesUrl + '\'' +
                ", keysUrl='" + keysUrl + '\'' +
                ", labelsUrl='" + labelsUrl + '\'' +
                ", languagesUrl='" + languagesUrl + '\'' +
                ", mergesUrl='" + mergesUrl + '\'' +
                ", milestonesUrl='" + milestonesUrl + '\'' +
                ", notificationsUrl='" + notificationsUrl + '\'' +
                ", pullsUrl='" + pullsUrl + '\'' +
                ", releasesUrl='" + releasesUrl + '\'' +
                ", sshUrl='" + sshUrl + '\'' +
                ", stargazersUrl='" + stargazersUrl + '\'' +
                ", statusesUrl='" + statusesUrl + '\'' +
                ", subscribersUrl='" + subscribersUrl + '\'' +
                ", subscriptionUrl='" + subscriptionUrl + '\'' +
                ", tagsUrl='" + tagsUrl + '\'' +
                ", teamsUrl='" + teamsUrl + '\'' +
                ", treesUrl='" + treesUrl + '\'' +
                '}';
    }
}

File: sample/src/main/java/com/smart/sample/domain/model/Trend.kt
package com.smart.sample.domain.model

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

data class Trend(
        @SerializedName("id")
        @Expose
        var id: Int? = null,
        @SerializedName("node_id")
        @Expose
        var nodeId: String? = null,
        @SerializedName("name")
        @Expose
        var name: String? = null,
        @SerializedName("full_name")
        @Expose
        var fullName: String? = null,
        @SerializedName("owner")
        @Expose
        var owner: Owner? = null,
        @SerializedName("private")
        @Expose
        var private: Boolean? = null,
        @SerializedName("html_url")
        @Expose
        var htmlUrl: String? = null,
        @SerializedName("description")
        @Expose
        var description: String? = null,
        @SerializedName("fork")
        @Expose
        var fork: Boolean? = null,
        @SerializedName("url")
        @Expose
        var url: String? = null,
        @SerializedName("archive_url")
        @Expose
        var archiveUrl: String? = null,
        @SerializedName("assignees_url")
        @Expose
        var assigneesUrl: String? = null,
        @SerializedName("blobs_url")
        @Expose
        var blobsUrl: String? = null,
        @SerializedName("branches_url")
        @Expose
        var branchesUrl: String? = null,
        @SerializedName("collaborators_url")
        @Expose
        var collaboratorsUrl: String? = null,
        @SerializedName("comments_url")
        @Expose
        var commentsUrl: String? = null,
        @SerializedName("commits_url")
        @Expose
        var commitsUrl: String? = null,
        @SerializedName("compare_url")
        @Expose
        var compareUrl: String? = null,
        @SerializedName("contents_url")
        @Expose
        var contentsUrl: String? = null,
        @SerializedName("contributors_url")
        @Expose
        var contributorsUrl: String? = null,
        @SerializedName("deployments_url")
        @Expose
        var deploymentsUrl: String? = null,
        @SerializedName("downloads_url")
        @Expose
        var downloadsUrl: String? = null,
        @SerializedName("events_url")
        @Expose
        var eventsUrl: String? = null,
        @SerializedName("forks_url")
        @Expose
        var forksUrl: String? = null,
        @SerializedName("git_commits_url")
        @Expose
        var gitCommitsUrl: String? = null,
        @SerializedName("git_refs_url")
        @Expose
        var gitRefsUrl: String? = null,
        @SerializedName("git_tags_url")
        @Expose
        var gitTagsUrl: String? = null,
        @SerializedName("git_url")
        @Expose
        var gitUrl: String? = null,
        @SerializedName("issue_comment_url")
        @Expose
        var issueCommentUrl: String? = null,
        @SerializedName("issue_events_url")
        @Expose
        var issueEventsUrl: String? = null,
        @SerializedName("issues_url")
        @Expose
        var issuesUrl: String? = null,
        @SerializedName("keys_url")
        @Expose
        var keysUrl: String? = null,
        @SerializedName("labels_url")
        @Expose
        var labelsUrl: String? = null,
        @SerializedName("languages_url")
        @Expose
        var languagesUrl: String? = null,
        @SerializedName("merges_url")
        @Expose
        var mergesUrl: String? = null,
        @SerializedName("milestones_url")
        @Expose
        var milestonesUrl: String? = null,
        @SerializedName("notifications_url")
        @Expose
        var notificationsUrl: String? = null,
        @SerializedName("pulls_url")
        @Expose
        var pullsUrl: String? = null,
        @SerializedName("releases_url")
        @Expose
        var releasesUrl: String? = null,
        @SerializedName("ssh_url")
        @Expose
        var sshUrl: String? = null,
        @SerializedName("stargazers_url")
        @Expose
        var stargazersUrl: String? = null,
        @SerializedName("statuses_url")
        @Expose
        var statusesUrl: String? = null,
        @SerializedName("subscribers_url")
        @Expose
        var subscribersUrl: String? = null,
        @SerializedName("subscription_url")
        @Expose
        var subscriptionUrl: String? = null,
        @SerializedName("tags_url")
        @Expose
        var tagsUrl: String? = null,
        @SerializedName("teams_url")
        @Expose
        var teamsUrl: String? = null,
        @SerializedName("trees_url")
        @Expose
        var treesUrl: String? = null
) {
//    override fun toString(): String {
//        return "Trend(id=$id, nodeId=$nodeId, name=$name, fullName=$fullName, owner=$owner, private=$private, htmlUrl=$htmlUrl, description=$description, fork=$fork, url=$url, archiveUrl=$archiveUrl, assigneesUrl=$assigneesUrl, blobsUrl=$blobsUrl, branchesUrl=$branchesUrl, collaboratorsUrl=$collaboratorsUrl, commentsUrl=$commentsUrl, commitsUrl=$commitsUrl, compareUrl=$compareUrl, contentsUrl=$contentsUrl, contributorsUrl=$contributorsUrl, deploymentsUrl=$deploymentsUrl, downloadsUrl=$downloadsUrl, eventsUrl=$eventsUrl, forksUrl=$forksUrl, gitCommitsUrl=$gitCommitsUrl, gitRefsUrl=$gitRefsUrl, gitTagsUrl=$gitTagsUrl, gitUrl=$gitUrl, issueCommentUrl=$issueCommentUrl, issueEventsUrl=$issueEventsUrl, issuesUrl=$issuesUrl, keysUrl=$keysUrl, labelsUrl=$labelsUrl, languagesUrl=$languagesUrl, mergesUrl=$mergesUrl, milestonesUrl=$milestonesUrl, notificationsUrl=$notificationsUrl, pullsUrl=$pullsUrl, releasesUrl=$releasesUrl, sshUrl=$sshUrl, stargazersUrl=$stargazersUrl, statusesUrl=$statusesUrl, subscribersUrl=$subscribersUrl, subscriptionUrl=$subscriptionUrl, tagsUrl=$tagsUrl, teamsUrl=$teamsUrl, treesUrl=$treesUrl)"
//    }

    override fun toString(): String {
        return "Trend(id=$id, name=$name)"
    }
}
